packages/gatsby/src/utils/dev-ssr/render-dev-html.ts

```diff
@@ -143,6 +145,27 @@ export const renderDevHTML = ({
       return reject(`404 page`)
     }
 
+    // Resume the webpack watcher and wait for any compilation necessary to happen.
+    // We timeout after 1.5s as the user might not care per se about SSR.
+    //
+    // We pause and resume so there's no excess webpack activity during normal development.
+    const { devssrWebpackCompilier, devssrWebpackWatcher } = getDevSSRWebpack()
+    if (devssrWebpackWatcher && devssrWebpackCompilier) {
+      await new Promise(resolve => {
+        function finish(stats: Stats): void {
+          emitter.off(`DEV_SSR_COMPILATION_DONE`, finish)
+          resolve(stats)
+        }
+        emitter.on(`DEV_SSR_COMPILATION_DONE`, finish)
+        devssrWebpackWatcher.resume()
+        // Suspending is just a flag, so it's safe to re-suspend right away
+        devssrWebpackWatcher.suspend()
+
+        // Timeout after 1.5s.
+        setTimeout(() => resolve(), 1500)
```

Comment:
this likely doesn't cause problems but we will call `resolve` twice because we don't clear timeout in `finish` callback / don't remove `DEV_SSR_COMPILATION_DONE` in timeout callback. So just hadling is a bit messy :)

Reply:
fixing!